src/linear_algebra/finite_dimensional.lean

```diff
@@ -54,44 +94,94 @@ begin
+maximal submodule, equal to the whole space as a set by definition.-/
+lemma iff_fg :
+  finite_dimensional K V ↔ (⊤ : submodule K V).fg :=
+begin
+  split,
+  { assume h,
+    haveI : finite_dimensional K V := h,
+    rcases exists_is_basis_finite K V with ⟨s, s_basis, s_finite⟩,
+    exact ⟨s_finite.to_finset, by { convert s_basis.2, simp }⟩ },
+  { rintros ⟨s, hs⟩,
+    rw [finite_dimensional_iff_dim_lt_omega, ← dim_top, ← hs],
+    exact lt_of_le_of_lt (dim_span_le _) (lt_omega_iff_finite.2 (set.finite_mem_finset s)) }
+end
+
+/-- If a vector space has a finite basis, then it is finite-dimensional. -/
+lemma of_finite_basis {ι : Type w} [fintype ι] {b : ι → V} (h : is_basis K b) :
+  finite_dimensional K V :=
+iff_fg.2 $ ⟨finset.univ.image b, by {convert h.2, simp} ⟩
+
+/-- A subspace of a finite-dimensional space is also finite-dimensional. -/
+instance finite_dimensional_submodule [finite_dimensional K V] (S : submodule K V) :
+  finite_dimensional K S :=
 finite_dimensional_iff_dim_lt_omega.2 (lt_of_le_of_lt (dim_submodule_le _) (dim_lt_omega K V))
 
-/-- The dimension of a finite-dimensional vector space as a natural number -/
+/-- A quotient of a finite-dimensional space is also finite-dimensional. -/
+instance finite_dimensional_quotient [finite_dimensional K V] (S : submodule K V) :
+  finite_dimensional K (quotient S) :=
+finite_dimensional_iff_dim_lt_omega.2 (lt_of_le_of_lt (dim_quotient_le _) (dim_lt_omega K V))
+
+/-- The dimension of a finite-dimensional vector space as a natural number. Defined by convention to
+be `0` if the space is infinite-dimensional. -/
 noncomputable def findim (K V : Type*) [discrete_field K]
-  [add_comm_group V] [vector_space K V] [finite_dimensional K V] : ℕ :=
-classical.some (lt_omega.1 (dim_lt_omega K V))
+  [add_comm_group V] [vector_space K V] : ℕ :=
+if h : dim K V < omega.{v} then classical.some (lt_omega.1 h) else 0
 
+/-- In a finite-dimensional space, its dimension (seen as a cardinal) coincides with its `findim`. -/
 lemma findim_eq_dim (K : Type u) (V : Type v) [discrete_field K]
   [add_comm_group V] [vector_space K V] [finite_dimensional K V] :
   (findim K V : cardinal.{v}) = dim K V :=
-(classical.some_spec (lt_omega.1 (dim_lt_omega K V))).symm
+begin
+  have : findim K V = classical.some (lt_omega.1 (dim_lt_omega K V)) :=
+    dif_pos (dim_lt_omega K V),
+  rw this,
+  exact (classical.some_spec (lt_omega.1 (dim_lt_omega K V))).symm
+end
 
-lemma card_eq_findim [finite_dimensional K V] {s : set V} {hfs : fintype s}
-  (hs : is_basis K (λ x : s, x.val)) : fintype.card s = findim K V :=
-by rw [← nat_cast_inj.{v}, findim_eq_dim, ← fintype_card, ← lift_inj, ← hs.mk_eq_dim]
+/-- If a vector space has a finite basis, then its dimension (seen as a cardinal) is equal to the
+cardinality of the basis. -/
+lemma dim_eq_card_basis {ι : Type w} [fintype ι] {b : ι → V} (h : is_basis K b) :
+  dim K V = fintype.card ι :=
+by rw [←h.mk_range_eq_dim, cardinal.fintype_card,
```

Comment:
Shouldn't this be the combination of some general `dim_eq_card_basis` (without any finiteness) and `fintype.card_eq_card` (comparing `fintype.card` with ordinary cardinals)?

Reply:
Yes, that's exactly what this proof does if I understand it correctly (I didn't write it, just moved it around).

Reply:
Ok... then it's just more work then I expected... as usual.